LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -156,17 +247,177 @@ List<Module> GetModuleList()
+            ReportFooter(action);
+
         }
-        
+
         public void AssignInstructorsToClassrooms()
         {
-            throw new NotImplementedException();
+            string action = "Assign instructor to classroom";
+            ReportHeader(action);
+
+            if (ActiveInstructorList.Count > 0 && ClassroomDictionary.Count > 0)
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    Guid moduleId = classroom.Module.Id;
+                    List<Instructor> instructorList;
+                    if (ModuleInstructorListDictionary.ContainsKey(moduleId))
+                    {
+                        ModuleInstructorListDictionary.TryGetValue(moduleId, out instructorList);
+                        classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
+                    }
+                }
+            }
+            else if (ActiveInstructorList.Count == 0)
+            {
+                ReportError(ActiveInstructorList.ToString(), action);
+                if (ClassroomDictionary.Count == 0)
+                {
+                    ReportError(ClassroomDictionary.ToString(), action);
+                }
+            }
+            ReportSummary(action, 1, "classroom");
+            ReportFooter(action);
+        }
+
+        void AddInstructorToModuleList(Instructor instructor)
+        {
+            List<Module> moduleList = instructor.ModuleList;
+            foreach (var module in moduleList)
+            {
+                List<Instructor> instructorList;
+                if (ModuleInstructorListDictionary.ContainsKey(module.Id))
+                {
+                    ModuleInstructorListDictionary.TryGetValue(module.Id, out instructorList);
+                    instructorList.Add(instructor);
+                }
+                else
+                {
+                    instructorList = new List<Instructor>() { instructor };
+                    ModuleInstructorListDictionary.Add(module.Id, instructorList);
+                }
+            }
         }
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            string action = "Register Students For Classes";
+            int itemListCount = 0;
+            ReportHeader(action);
+
+            if (ClassroomDictionary.Count > 0)
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        itemListCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+            else
+            {
+                ReportError(ClassroomDictionary.ToString(), action);
+            }
+
+            ReportSummary(action, itemListCount, "student");
```

Comment:
You summary statement is incorrect. You should report here for how many classes you registered students.